### PR TITLE
feat: improve mm impersonator checks

### PR DESCRIPTION
**Pull request by gomesalexandre**
## Description

Adds a "definitive" (for now at least) list of MM impersonators and explicitly checks against them with a new `checkIsMetaMaskImpersonator` method in addition to our `checkIsMetaMask` method, to avoid impersonators being detected as MM.

## Pull Request Type

- [ ] :bug: Bug fix (Non-breaking Change: Fixes an issue)
- [ ] :hammer_and_wrench: Chore (Non-breaking Change: Doc updates, pkg upgrades, typos, etc..)
- [x] :nail_care: New Feature (Breaking/Non-breaking Change)

## Issue (if applicable)

- losely relates to https://github.com/shapeshift/web/issues/5138

## Risk

- MM could be wrongly detected as an impersonator and the snap modal may not be displayed for it, test according to the testing steps

## Testing

- When connecting Rabby wallet (or any of the wallets in the `METAMASK_IMPERSONATORS` e.g Zerion wallet extension), the MM snaps modal doesn't show up
- When connecting *actual* MM, the MM snaps modal shows up
- Connecting previously not working MM impersonators (e.g Zerion) is now possible. Note, this is not a magic solution that will make *any* impersonator work as they all have their ramifications. Tested Zerion is now happy after https://github.com/shapeshift/hdwallet/pull/657, but if other impersonators are still not working, this will need to be tackled as a per-wallet basis.

### Engineering

- ☝🏽 

### Operations

- ☝🏽 

### MM snaps modal only shows up for actual MM

- Develop 🚫  - modal shows up for MM, but also impersonators

image
image
Screenshot 2023-12-19 at 13 10 10
Screenshot 2023-12-19 at 13 10 17

- This diff ✅ - modal shows up only for *actual* MM

image
image
image

### CODE REVIEW

**Comment by gomesalexandre** [comment]
Current dependencies on/for this PR:
* `develop`
  * **PR #5896** <a href="https://app.graphite.dev/github/pr/shapeshift/web/5896?utm_source=stack-comment-icon" target="_blank">Graphite</a>  👈

This [stack of pull requests](https://stacking.dev/?utm_source=stack-comment) is managed by [Graphite](https://app.graphite.dev/github/pr/shapeshift/web/5896?utm_source=stack-comment).

File: .yarnrc.yml [hunk 1]
```diff
@@ -1,6 +1,6 @@
 nodeLinker: node-modules
 
-npmRegistryServer: "https://registry.npmjs.org"
+npmRegistryServer: "http://127.0.0.1:4873/"
```
Comment (gomesalexandre) [inline .yarnrc.yml:3]: Revert before opening

File: package.json [hunk 2]
```diff
@@ -87,19 +87,19 @@
...
-    "@shapeshiftoss/hdwallet-keplr": "1.52.14",
-    "@shapeshiftoss/hdwallet-ledger": "1.52.14",
-    "@shapeshiftoss/hdwallet-ledger-webusb": "1.52.14",
-    "@shapeshiftoss/hdwallet-metamask": "1.52.14",
-    "@shapeshiftoss/hdwallet-native": "1.52.14",
-    "@shapeshiftoss/hdwallet-native-vault": "1.52.14",
-    "@shapeshiftoss/hdwallet-shapeshift-multichain": "1.52.14",
-    "@shapeshiftoss/hdwallet-walletconnectv2": "1.52.14",
-    "@shapeshiftoss/hdwallet-xdefi": "1.52.14",
+    "@shapeshiftoss/hdwallet-coinbase": "1.52.15-alpha.7",
```
Comment (gomesalexandre) [inline package.json:90]: Install actual 1.52.15 when published

Comment (0xApotheosis) [reply package.json]: Actioned in https://github.com/shapeshift/web/pull/5896/commits/cf7d5c99a8aa0537ef7a53cf4a905913af086498.

File: .yarnrc.yml (hunk 1 above)
Comment (0xApotheosis) [reply .yarnrc.yml]: Actioned in https://github.com/shapeshift/web/pull/5896/commits/cf7d5c99a8aa0537ef7a53cf4a905913af086498.

**Review by 0xApotheosis — APPROVED** [review]
I'm getting a snaps install prompt with using actual metamask, but not when using Brave Wallet 👌

**Pull request merged**